libraries/AP_OpenDroneID/AP_OpenDroneID.cpp

```diff
@@ -147,6 +206,16 @@ void AP_OpenDroneID::update()
         return;
     }
 
+    if ((pkt_basic_id.id_type == MAV_ODID_ID_TYPE_SERIAL_NUMBER)
+        && (_options & LockUASIDOnFirstBasicIDRx)
+        && id_len == 0) {
+        hal.util->flash_bootloader();
```

Comment:
I think we need a stronger mechanism to prevent calling this flash_bootloader() more than once per boot. The scenario I'm thinking of is if id_len comes back as zero due to another error we would end up flashing the bootloader continuously.
The simple fix is a bool done_flash_bootloader in this class, and if this is true then don't flash again

Reply:
Done.